### PR TITLE
Fsx for lustre added as module to Karpenter example

### PR DESCRIPTION
**Pull request by vara-bonthu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

- RAID0 config added to Karpenter examples
- FSx for Lustre as a sub module for Karpenter. FSx for Lustre blueprint will be removed later
- Workshop folder structure created

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/awslabs/data-on-eks/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR to add a new Add-on for [Terraform EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) repo (if applicable)
- [ ] Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-static-pvc.yaml
Comment (dalbhanj) [inline analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-static-pvc.yaml:1]: Same here...

File: analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-storage-class.yaml
Comment (dalbhanj) [inline analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-storage-class.yaml:1]: We can probably get away with this one as I understand FSx volume gets created when you create a PVC

File: analytics/terraform/emr-eks-karpenter/variables.tf
Comment (dalbhanj) [inline analytics/terraform/emr-eks-karpenter/variables.tf:1]: This looks good and provides better experience for customers!

**Review by dalbhanj — CHANGES_REQUESTED** [review]
I left comment on create FSx piece of infrastructure separate and available on-demand when users want to use it rather than part of the whole infrastructure setup. Please take a look. Thanks!

File: analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-dynamic-pvc.yaml
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-dynamic-pvc.yaml:1]: This is now a separate module

File: analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-static-pvc.yaml
Comment (vara-bonthu) [reply analytics/terraform/emr-eks-karpenter/modules/fsx-for-lustre/fsx_lustre/fsxlustre-static-pvc.yaml:1]: updated

**Pull request merged**